### ISSUE TITLE
`cargo build --verbose` command output does not quote rustc arguments

### ISSUE THREAD

**Issue by jmesmon** [rust-lang/cargo#744]
This makes it impossible to determine exactly what the rustc arguments were, especially when they contain `-C linker` with some flags.

Example:
```
 Running `rustc /home/cody/.cargo/git/checkouts/git2-rs-cf258e7bcbaee34d/master/libgit2/src/lib.rs --crate-name libgit2 --crate-type lib -g -C metadata=fcba23a36b59f31c -C extra-filename=-fcba23a36b59f31c --out-dir /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/deps --dep-info /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/.fingerprint/libgit2-fcba23a36b59f31c/dep-lib-libgit2 --target x86_64-unknown-linux-gnu -C ar=ar -C linker=gcc -L/home/cody/obj/y/tmp/sysroots/x86_64-linux/usr/lib -L/home/cody/obj/y/tmp/sysroots/x86_64-linux/lib -Wl,-rpath-link,/home/cody/obj/y/tmp/sysroots/x86_64-linux/usr/lib -Wl,-rpath-link,/home/cody/obj/y/tmp/sysroots/x86_64-linux/lib -Wl,-rpath,/home/cody/obj/y/tmp/sysroots/x86_64-linux/usr/lib -Wl,-rpath,/home/cody/obj/y/tmp/sysroots/x86_64-linux/lib -Wl,-O1 -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/deps -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/deps -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/native/libgit2-fcba23a36b59f31c -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/native/openssl-static-sys-6ae299ea46a69e57 -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/native/libssh2-static-sys-c29829f2cf163ffd --extern link-config=/home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/deps/liblink-config-3e5e46ed95de9618.so --extern openssl-static-sys=/home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/deps/libopenssl-static-sys-6ae299ea46a69e57.rlib --extern libssh2-static-sys=/home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/deps/liblibssh2-static-sys-c29829f2cf163ffd.rlib`
```

In this case, linker supplies a number of -L and -Wl,-rpath-link options.

**Comment by bluss** [comment]
Which argument needs quoting here? I couldn't find which is intended. I assume there is a space or a special character somewhere.

**Comment by jmesmon** [comment]
`-C linker[...] -L /home/cody/obj/y/tmp/work/x86_64-linux/cargo-native/2014-10-09-r0/git/target/x86_64-unknown-linux-gnu/native/libssh2-static-sys-c29829f2cf163ffd `

Spaces need the be quoted. I had a PR out that did this: https://github.com/rust-lang/rust/pull/19441 but it was closed because it made the debug output less pretty.

**Comment by bluss** [comment]
I'm working on it. I didn't see your issue or PR before I posted mine.

**Comment by alexcrichton** [comment]
Fixed in #1770
